pylzy/lzy/api/v1/remote/runtime.py

```diff
@@ -340,6 +347,23 @@ def __build_graph(
                 )
                 ret_descriptions.append((entry.typ, slot_path))
 
+            entry = self.__workflow.snapshot.get(call.exception_id)
```

Comment:
I use it the same way as any other snapshot entries, it is useful for downloading data the same way for example.

Reply:
I don't think it is a good reason to write bad code. Maybe it is a good reason to rewrite other places like this.
Look, below you have this code:
```
            exc_entry = self.__workflow.snapshot.get(call.exception_id)
            exc_slot_path = f"/{call.id}/{exc_entry.id}"
            output_slots.append(Operation.SlotDescription(path=exc_slot_path, storageUri=exc_entry.storage_uri))
            data_descriptions[exc_entry.storage_uri] = data_description_by_entry(exc_entry)
            exc_description: Tuple[Type, str] = (exc_entry.typ, exc_slot_path)
```
in a middle of build graph function.
How much better it would be if we could rewrite it next way?
```
exception_info = self._workflow.exception_info
slot_description = Operation.SlotDescription(path=exception_info.slot_path, storageUri=exception_info.storage_uri)
data_descriptions[exception_info.storage_uri]  = data_description_by_entry(exception_info.snapshot_entry)
```
and now use `exception_info` instead of `exc_description`?

This would be a convenient data structure with incapsulated information about exception, so you wouldn't restore from snapshot every time you need it, every time build paths in different areas of code, etc.

Reply:
1) This paths and types are used only here (and they are different for local and remote runtime), so there is no need to store them in workflow. Everywhere else we need only ids of entries to get data, for example.
2) I am not saying, that code is good =)
But it is consistent with already written code. If the structure is bad, let's create a ticket and refactor this in different PR. For example, create new data class instead of tuple.